### ISSUE TITLE
Improve the definition of "Content"

### ISSUE THREAD

**Issue by DuffJohnson** [w3c/wcag#4773]
WCAG defines "[Content](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-content)" as:

> information and sensory experience to be communicated to the user by means of a [user agent](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-user-agent), including code or markup that defines the content's [structure](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-structure), [presentation](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-presentation), and interactions

It has been [explained](https://github.com/w3c/wcag/issues/4655#issuecomment-3543431570) that this definition is not intended to suggest that markup is required. I believe that the definition of content should explicitly NOT include markup, which should be defined distinctly from content... but that's another subject.

Based on the current definition's intent a better formulation would appear to be (emphasis applied to the proposed added words):

> information and sensory experience to be communicated to the user by means of a [user agent](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-user-agent), including code or markup, **if any**, that defines the content's [structure](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-structure), [presentation](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-presentation), and interactions

**Comment by yatil** [comment]
“Including” already indicates that markup is not required, but that it can be part of content. I don’t think “if any” clarifies the definition.

**Comment by DuffJohnson** [comment]
So, does content without any "code or markup" have a "defined structure", as would be required to satisfy SC 1.3.1.?

What about markup without "content" (that is, the stuff inside the tag)? Is this also "content"? According to some discussing #4655 the answer is "yes"... but WCAG's definition doesn't regard markup as content in-itself - markup is instead something that may be (but isn't necessarily) "included" with content.

If clarifying this definition is unpalatable for now, perhaps a future WCAG could include distinct definitions for "content" and "markup".

**Comment by GreggVan** [comment]
@DuffJohnson 

If I said "all people, including children"  you would not think that children were not people.  just a clarification that they were included in the definition of people. 

Similarly,  "content including markup" does not imply in any way that markup is not content.    One could define markup -- but it would not be separate from but a subset of content.   We don't usually define single words that are used as they are normally defined and know -- but could pull a definition from someplace for convenience of non-technical readers. 

RE can a document without markup have structure....   the answer is absolutely.

**Comment by DuffJohnson** [comment]
> If I said "all people, including children" you would not think that children were not people. just a clarification that they were included in the definition of people.

I don't think that analogy fits. "Markup is to content as professions are to people" seems more accurate to me.

> Similarly, "content including markup" does not imply in any way that markup is not content. 

I'm simply pointing out that as we discuss content without markup and markup without content, its useful to conceptually distinguish content from markup, and that the current definition of "content" doesn't help in this regard.

> One could define markup -- but it would not be separate from but a subset of content. 

That doesn't accord with the way markup is discussed in WCAG. See, for example, the description given in [G115](https://www.w3.org/WAI/WCAG21/Techniques/general/G115
), in which markup is presented (correctly, IMO) as something that operates on content.

> We don't usually define single words that are used as they are normally defined and know -- but could pull a definition from someplace for convenience of non-technical readers.

I'm curious to see what you find. [Wikipedia's entry for "markup language"](https://en.wikipedia.org/wiki/Markup_language
) appears clear enough to me.

> RE can a document without markup have structure.... the answer is absolutely.

If so, I don't think this is obvious from WCAG's definition of "content", which appears to suggest that the only mechanisms that provide structure are "code or markup".

**Comment by patrickhlauke** [comment]
I admit I'm failing to see the exact distinction the "if any" is trying to make here.

Admittedly the recent discussion on empty headings https://github.com/w3c/wcag/issues/4655#issuecomment-3543431570 was a weird (and controversial, as there was far from unanimous agreement on that take) was a weird edge case, but I don't think the addition of "if any" or whatever would have made it any clearer. Conceptually, that discussion was probably more an indication that we can't look at pure content *without the markup/code that conveys it* in isolation. The markup/code materialises the programmatic structure inside the user agent, which is then used by the user agent to convey the raw content, and in doing so imparts structure to the content. in the case of the empty heading discussion, while the "raw" content didn't have a heading there, just an empty line, in concert with the underlying markup, the user agent "transformed" that empty line internally into a heading which it then conveyed to the user. it's the mediated content structure that was problematic. that's not about "does markup count as content".

> can a document without markup have structure.... the answer is absolutely

it may be the case that there is some intended intrinsic structure to content that isn't marked up at all. but that structure won't be *programmatically* defined, which is the problem 1.3.1 Info and Relationships etc try to tackle. as in: I could write something in pure plain text, and structure it so that there are headings, bullet points, block quotes, etc by virtue of just adding extra empty lines of text, asterisks for bullets, etc. the content here has "structure" even without markup. but it's not in a way that can be *programmatically determined* (without, for instance, some tool going over that not-programmatically-structured content and using heuristics to infer "this is likely a heading, so let me expose it as a heading programmatically", "this set of lines all prefixed with an asterisk are likely to be bullet points in an unordered list, so let me expose them as such")

I'll admit this is all starting to get a bit metaphysical and zen buddhist ("...and if content falls in the woods and there's no user agent there to see it...")

**Comment by DuffJohnson** [comment]
> I admit I'm failing to see the exact distinction the "if any" is trying to make here.
>
> Admittedly the recent discussion on empty headings [#4655 (comment)](https://github.com/w3c/wcag/issues/4655#issuecomment-3543431570) was a weird (and controversial, as there was far from unanimous agreement on that take) was a weird edge case

Well... markup that doesn't enclose any "raw content" (to borrow your excellent term) may be an edge-case in HTML, but it's commonplace (sadly) in PDF and elsewhere. I got involved in #4655 owning to this fact, as it seems unreasonable to fail a document due to "empty markup", which was the question animating that issue.

> , but I don't think the addition of "if any" or whatever would have made it any clearer.

I agree that changing the definition of Content would not affect the resolution of #4655. This issue (#4773) arose from my [convo with @GreggVan on #4655](https://github.com/w3c/wcag/issues/4655#issuecomment-3543431570), and was acknowledged to be a side-issue, which is why I opened a new issue here.

As I hope I've made clear, I don't think the current definition of "content" is a good one. Instead, I've proposed that some future WCAG adopt distinct definitions for "content" and "markup". My suggestion to add "if any" to the existing definition was simply an attempt to bring it more in line with Gregg's explanation of the (current) definition's intent.

> Conceptually, that discussion was probably more an indication that we can't look at pure content _without the markup/code that conveys it_ in isolation. The markup/code materialises the programmatic structure inside the user agent, which is then used by the user agent to convey the raw content, and in doing so imparts structure to the content. in the case of the empty heading discussion, while the "raw" content didn't have a heading there, just an empty line, in concert with the underlying markup, the user agent "transformed" that empty line internally into a heading which it then conveyed to the user. it's the mediated content structure that was problematic. that's not about "does markup count as content".

IMO, the fact that you are using "raw content" to make this point helps explain why a definitional distinction between "content" and "markup" would be useful.

In this context it seems relevant to mention that In PDF we are obliged to distinguish between:

- "real content" (semantic markup is provided), 
- "artifact content" (marked as one of various flavors of non-semantic content), and
- "unmarked content" (lacks any semantic _or_ artifact markup) 
 
 ...but I prefer "raw content" to "unmarked"!

> > can a document without markup have structure.... the answer is absolutely
> 
> it may be the case that there is some intended intrinsic structure to content that isn't marked up at all. but that structure won't be _programmatically_ defined, which is the problem 1.3.1 Info and Relationships etc try to tackle. as in: I could write something in pure plain text, and structure it so that there are headings, bullet points, block quotes, etc by virtue of just adding extra empty lines of text, asterisks for bullets, etc. the content here has "structure" even without markup. but it's not in a way that can be _programmatically determined_ (without, for instance, some tool going over that not-programmatically-structured content and using heuristics to infer "this is likely a heading, so let me expose it as a heading programmatically", "this set of lines all prefixed with an asterisk are likely to be bullet points in an unordered list, so let me expose them as such")

+1
